text/1162449/undirected_ring_stages_with_feedback.tex

```diff
@@ -0,0 +1,99 @@
+\documentclass{article}
+
+\usepackage[english]{babel}
+\usepackage{filecontents}
+\usepackage[letterpaper,top=2cm,bottom=2cm,left=3cm,right=3cm,marginparwidth=1.75cm]{geometry}
+
+\usepackage{amsmath}
+\usepackage{csquotes}
+\usepackage{graphicx}
+\usepackage{amsthm}
+\usepackage[colorlinks=true, allcolors=blue]{hyperref}
+\begin{filecontents}{\jobname.bib}
+@article{ROTEM1987575,
+title = {Analysis of a distributed algorithm for extrema finding in a ring},
+journal = {Journal of Parallel and Distributed Computing},
+volume = {4},
+number = {6},
+pages = {575-591},
+year = {1987},
+issn = {0743-7315},
+doi = {https://doi.org/10.1016/0743-7315(87)90031-1},
+url = {https://www.sciencedirect.com/science/article/pii/0743731587900311},
+author = {D. Rotem and E. Korach and N. Santoro},
+abstract = {A new and more detailed analysis of the unidirectional algorithm of Chang and Roberts for distributed extrema finding on a ring is presented. This analysis shows that this simple algorithm, which is known to be average case optimal, compares very favorably with all the other known algorithms as it requires O(n log n) messages with probability tending to one. A bidirectional version of this algorithm is presented and is shown to dominate the unidirectional one in its average message complexity. Finally, both the unidirectional and the bidirectional algorithms are generalized to perform k selection in the ring, i.e., find the k largest labeled processors.}
+}
+\end{filecontents}
+
+\usepackage{natbib}
+\newtheorem*{lemma}{Lemma}
+\newtheorem*{theorem}{Theorem}
+\begin{document}
+\section*{Leader election in an undirected ring}
+Our goal is to elect a leader in an undirected ring with $n$ nodes and asynchronous first-in-first-out communication.  We assume that every node has a unique identifier, and that all identifiers are
+totally ordered. Initially, no node needs any knowledge about the identities of the nodes at the
+other ends of its incident edges. Each node recognizes the directions of its edges, but there is no global sense of orientation in the ring. 
+\paragraph{}
+Here we describe an algorithm presented by \cite{ROTEM1987575}, which achieves the above and uses at most $3 n\log_{3}n + O(n)$ messages in the worst case.
+\subsection*{The algorithm}
+The algorithm will consist of electoral rounds.\\
+We will distinguish 4 states a node can be in:\\
+\textit{passive} - node does not participate in election anymore and only passes messages\\
+(node that is not "\textit{passive}" will be also regarded as "\textit{active}")\\
+\textit{E-candidate} - node is taking part in this electoral round\\ \textit{A-candidate} - node awaits feedback from its active neighbours\\ \textit{Leader} - node is the leader\\
```

Comment:
Single - is "dywiz" (as in biało-czerwony). Here you need -- (półpauza) or --- (pauza), as this is the one surrounded by spaces. See also https://pl.wikipedia.org/wiki/Dywiz